packages/runtime-core/src/hydration.ts

```diff
@@ -365,7 +365,12 @@ export function createHydrationFunctions(
     const forcePatch = type === 'input' || type === 'option'
     // skip props & children if this is hoisted static nodes
     // #5405 in dev, always hydrate children for HMR
-    if (__DEV__ || forcePatch || patchFlag !== PatchFlags.HOISTED) {
+    if (
+      __DEV__ ||
+      __FEATURE_PROD_HYDRATION_MISMATCH_DETAILS__ ||
```

Comment:
Why forcing hoisted static nodes to be hydrated when this flag is on?

Reply:
Thanks. Just tested that it's not needed to show the text content mismatch warnings. Removed it from the condition.